### PR TITLE
Ml api integration

### PR DESCRIPTION
**Pull request by ezronis**

### CODE REVIEW

File: match/admin.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from django.contrib import admin
+from django.conf.urls import url, include
+from django.urls import path
+from match.models import *
+#
+# class MatchesAdmin(admin.ModelAdmin):
```
Comment (jremmanue) [inline match/admin.py]: i would remove commented out code

File: match/admin.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    def get_urls(self):
+        urls = super().get_urls()
+        my_urls = [
+            path('immortal/', self.set_immortal),
+            path('mortal/', self.set_mortal),
+        ]
+        return my_urls + urls
+
+    def set_immortal(self, request):
```
Comment (jremmanue) [inline match/admin.py]: i would clean this up so variables make sense

File: match/templates/buttons.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div>
+        <form action="immortal/" method="POST">
```
Comment (jremmanue) [inline match/templates/buttons.html]: again maybe change immortal/mortal thing to contextualize into app

File: ml/mentor_match_classifier/mentor_match.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    def postprocessing(self, prediction, input_data):
+        match_json_result = self.mm.matches_to_json(input_data, prediction)
+        return {"matches": match_json_result, "status": "OK"}
+
+    def compute_prediction(self, input_data):
+        try:
+            input_data = self.preprocessing(input_data)
+            prediction = self.predict(input_data)
+            prediction = self.postprocessing(prediction, input_data)
+            print(prediction)
```
Comment (jremmanue) [inline ml/mentor_match_classifier/mentor_match.py]: remove print statment

File: ml/mentor_match_classifier/mentor_match_utils.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    def assign_id(self, surveys, score_matrix): 
+        mentor_subset_df = surveys[surveys['Are you a Mentor or Student?']== 'mentor']
+        mentor_subset_df.reset_index(drop=True, inplace=True)  #only thing is how to gain access to the df here 
+        score_matrix['mentor_id'] = mentor_subset_df['ID']
+        score_matrix.set_index('mentor_id', inplace = True)
+        return score_matrix 
+
+    def final_score(self, region, no_group, with_group):
+        # if(region == 0):     #if the mentor/mentee have dif regions
```
Comment (jremmanue) [inline ml/mentor_match_classifier/mentor_match_utils.py]: remove commented out

File: survey/exporter/csv/survey2csv.py [hunk 6]
```diff
@@ -69,10 +71,12 @@ def get_header_and_order(self):
 
         :param Survey survey: The survey we're treating. """
         header = [_("user")]  # , u"entity"]
+        # header += [_("user_id")]
```
Comment (jremmanue) [inline survey/exporter/csv/survey2csv.py]: remove line

File: survey/exporter/csv/survey2csv.py [hunk 7]
```diff
@@ -89,6 +93,9 @@ def __str__(self):
     @staticmethod
     def export_as_csv(modeladmin, request, queryset):
         response = HttpResponse(content_type="text/csv")
+        print(response)
```
Comment (jremmanue) [inline survey/exporter/csv/survey2csv.py]: remove print statements

**Review by jremmanue — CHANGES_REQUESTED** [review]

File: userProfile/admin.py [hunk 8]
```diff
@@ -17,8 +17,22 @@ class AdditionalQuestionsAdmin(nested_admin.NestedModelAdmin):
...
    inlines = [QuestionInline,]
  
+# class ResponseInline(admin.TabularInline):
+#    model = Response
+#    def formfield_for_foreignkey(self,db_field , request, **kwargs):
+#       list_display = (self.model.question,)
+#       if db_field.name == "answer":
+#          kwargs["queryset"] = Answer.objects.filter(question_id = list_display[0].id)
+#       return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
```
Comment (ezronis) [inline userProfile/admin.py:28]: @amaliatorres2 are these changes still valid?

File: userProfile/models.py [hunk 9]
```diff
@@ -70,7 +69,7 @@ class Response(models.Model):
     question_form = models.ForeignKey(AdditionalQuestions, on_delete=models.CASCADE, null=True)
     user = models.ForeignKey(Profile, on_delete=models.CASCADE, null = True)
     question = models.ForeignKey(Question, on_delete=models.CASCADE, null=True)
-    answer = models.ForeignKey(Answer,on_delete=models.CASCADE,null=True)
+    answer = models.ForeignKey(Answer,on_delete=models.CASCADE, limit_choices_to=Q(question_id = 1), null=True)
```
Comment (ezronis) [inline userProfile/models.py]: @amaliatorres2 are these changes still valid?

File: userProfile/models.py [hunk 10]
```diff
@@ -29,9 +29,8 @@ def __str__(self):
         return self.label
 
 class Answer(models.Model):
-    question = models.ForeignKey(Question, on_delete=models.CASCADE)
+    question = models.ForeignKey(Question, on_delete=models.CASCADE, related_name='answers')
```
Comment (ezronis) [inline userProfile/models.py]: @amaliatorres2 are these changes still valid?

File: userProfile/templates/pages/edit_profile.html [hunk 11]
```diff
@@ -137,6 +137,12 @@ <h4>{{user.profile.first_name}}'s Profile</h4>
 
         {% endfor %}
       </div>
+      {% for field in add_form%}
```
Comment (ezronis) [inline userProfile/templates/pages/edit_profile.html]: @amaliatorres2 are these changes still valid?

File: userProfile/views.py [hunk 12]
```diff
@@ -79,18 +79,22 @@ def edit_profile(request):
...
             return redirect("profile")
         else:
             messages.error(request, _('Please correct the error below.'))
     else:
         profile_form = ProfileForm(instance=request.user.profile)
+        # add_form = AdditionalQuestionsForm(instance=request.user.profile)
     return render(request, 'pages/edit_profile.html', {
-
-        'profile_form': profile_form
+        'profile_form': profile_form,
```
Comment (ezronis) [inline userProfile/views.py]: @amaliatorres2 are these changes still valid?

File: userProfile/models.py (hunk 9 above)
Comment (amaliatorres) [reply userProfile/models.py]: no

File: userProfile/models.py (hunk 10 above)
Comment (amaliatorres) [reply userProfile/models.py]: no

File: userProfile/templates/pages/edit_profile.html (hunk 11 above)
Comment (amaliatorres) [reply userProfile/templates/pages/edit_profile.html]: no

File: userProfile/views.py (hunk 12 above)
Comment (amaliatorres) [reply userProfile/views.py]: no

**Pull request merged**